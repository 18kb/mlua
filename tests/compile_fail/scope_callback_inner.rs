use mlua::{Lua, Table, Result};

struct Test {
    field: i32,
}

fn main() {
    let lua = Lua::new();
    lua.scope(|scope| -> Result<()> {
        let mut inner: Option<Table> = None;
        let f = scope
            .create_function_mut(|_, t: Table| {
                inner = Some(t);
                Ok(())
            })?;
        f.call::<_, ()>(lua.create_table()?)?;
        Ok(())
    });
}
