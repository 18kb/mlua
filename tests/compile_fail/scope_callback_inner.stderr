error[E0495]: cannot infer an appropriate lifetime for autoref due to conflicting requirements
  --> $DIR/scope_callback_inner.rs:12:14
   |
12 |             .create_function_mut(|_, t: Table| {
   |              ^^^^^^^^^^^^^^^^^^^
   |
note: first, the lifetime cannot outlive the anonymous lifetime #2 defined on the body at 9:15...
  --> $DIR/scope_callback_inner.rs:9:15
   |
9  |       lua.scope(|scope| -> Result<()> {
   |  _______________^
10 | |         let mut inner: Option<Table> = None;
11 | |         let f = scope
12 | |             .create_function_mut(|_, t: Table| {
...  |
17 | |         Ok(())
18 | |     });
   | |_____^
note: ...so that reference does not outlive borrowed content
  --> $DIR/scope_callback_inner.rs:11:17
   |
11 |         let f = scope
   |                 ^^^^^
note: but, the lifetime must be valid for the method call at 9:5...
  --> $DIR/scope_callback_inner.rs:9:5
   |
9  | /     lua.scope(|scope| -> Result<()> {
10 | |         let mut inner: Option<Table> = None;
11 | |         let f = scope
12 | |             .create_function_mut(|_, t: Table| {
...  |
17 | |         Ok(())
18 | |     });
   | |______^
note: ...so that a type/lifetime parameter is in scope here
  --> $DIR/scope_callback_inner.rs:9:5
   |
9  | /     lua.scope(|scope| -> Result<()> {
10 | |         let mut inner: Option<Table> = None;
11 | |         let f = scope
12 | |             .create_function_mut(|_, t: Table| {
...  |
17 | |         Ok(())
18 | |     });
   | |______^
